release-tools/prow.sh

```diff
@@ -137,14 +137,23 @@ configvar CSI_PROW_BUILD_JOB true "building code in repo enabled"
 configvar CSI_PROW_KUBERNETES_VERSION 1.17.0 "Kubernetes"
 
 # This is a hack to workaround the issue that each version
-# of kind currently only supports specific patch versions of
-# Kubernetes. We need to override CSI_PROW_KUBERNETES_VERSION
+# of kind currently only has images for specific patch versions of
+# Kubernetes, as documented in the release notes of each
+# kind release.
+#
+# We need to override CSI_PROW_KUBERNETES_VERSION
 # passed in by our CI/pull jobs to the versions that
-# kind v0.5.0 supports.
+# kind v0.9.0 supports (https://github.com/kubernetes-sigs/kind/releases/tag/v0.9.0).
 #
 # If the version is prefixed with "release-", then nothing
 # is overridden.
-override_k8s_version "1.15.3"
+override_k8s_version "1.13.12"
+override_k8s_version "1.14.10"
+override_k8s_version "1.15.12"
+override_k8s_version "1.16.15"
+override_k8s_version "1.17.11"
+override_k8s_version "1.18.8"
+override_k8s_version "1.19.1"
```

Comment:
Overriding the K8S version was added in https://github.com/kubernetes-csi/external-provisioner/pull/313 because KinD only has images for certain versions.

But the recommendations in the KinD release notes go even further than that: it can happen that `kindest/node:v1.19.1` becomes incompatible with kind v0.9 if the image needs to be rebuilt for, e.g. kind 1.0.

@msau42: should we switch to using images with hash, i.e. `kindest/node:v1.19.1@sha256:98cf5288864662e37115e362b23e4369c8c4a408f99cbc06e58ac30ddc721600`?

Reply:
Sure let's use images with hash.

But I also want to move away from the model of specifying versions in prow.sh, so that we don't need to update prow.sh every time we need to add a new K8s version (although maybe unavoidable until kind apis stabilze more). I prefer to put the burden on the caller to make sure the kind version + k8s version specified are compatible.